server/datastore/mysql/policies.go

```diff
@@ -682,6 +682,57 @@ func (ds *Datastore) CleanupPolicyMembership(ctx context.Context, now time.Time)
 	return nil
 }
 
+func (ds *Datastore) UpdatePolicyFailureCountsForHosts(ctx context.Context, hosts []*fleet.Host) ([]*fleet.Host, error) {
+	// Get policy failure counts for each host
+	hostIDs := make([]uint, 0, len(hosts))
+
+	for _, host := range hosts {
+		hostIDs = append(hostIDs, host.ID)
+	}
+
+	query, args, err := sqlx.In(`
+		SELECT
+			pm.host_id,
+			COUNT(*) AS failing_policy_count
+		FROM
+			policy_membership pm
+		INNER JOIN
+			policies p
```

Comment:
Do we really need to `INNER JOIN policies p`? (We have a foreign key relationship already so AFAICS it's not needed.)
https://github.com/fleetdm/fleet/blob/7f19069f38beb73b9e805c2e730ee9276fae61eb/server/datastore/mysql/schema.sql#L1023